File: check-if-a-string-contains-all-binary-codes-of-size-k/check-if-a-string-contains-all-binary-codes-of-size-k.java
class Solution {
    public boolean hasAllCodes(String s, int k) {
        
        HashSet<String> codes = new HashSet();
        
        for (int i = 0; i < s.length() - k + 1; ++i) {
            
            codes.add(s.substring(i, i + k));
        }
        
        return (codes.size() == Math.pow(2, k));
    }
}

File: check-if-a-string-contains-all-binary-codes-of-size-k/check-if-a-string-contains-all-binary-codes-of-size-k.cpp
class Solution {
public:
    bool hasAllCodes(string s, int k) {
        
        if (s.size() < k) { return false; }
        
        set<std::string> codes {};
        
        for (int i = 0; i < s.size() - k + 1; ++i) {
            
            codes.insert(s.substr(i, k));
        }
        
        return (codes.size() == pow(2, k));
    }
};